apps/extension/src/routes/popup/approval/tx-msg-handler.ts

```diff
@@ -10,7 +10,8 @@ export const isTxApprovalReq = (req: PopupRequest): req is TxApproval => {
 export const handleTxApproval: InternalMessageHandler<TxApproval> = (jsonReq, responder) => {
   useStore.setState(state => {
     // @ts-expect-error Typescript doesn't like JsonValue could possibly be very deep
-    state.txApproval.tx = jsonReq;
+    state.txApproval.authorizeRequest = jsonReq.authorizeRequest;
+    state.txApproval.denomMetadataByAssetId = jsonReq.denomMetadataByAssetId;
```

Comment:
it might be reasonable to do something like AuthorizeRequest.fromJson() here to 'deserialize'

Reply:
We actually never ended up needing the JSON value of `AuthorizeRequest` — we just display it in a JSONViewer.